curriculum/challenges/english/25-front-end-development/lab-user-configuration-manager/684aaf9ec670c68d20efd0d0.md

```diff
@@ -0,0 +1,384 @@
+---
+id: 684aaf9ec670c68d20efd0d0
+title: Build a User Configuration Manager
+challengeType: 27
+dashedName: build-a-user-configuration-manager
+---
+
+# --description--
+
+Fulfill the user stories below and get all the tests to pass to complete the lab.
+
+**User Stories:**
+
+1. You should create a dictionary named `settings` that would store the following user configuration preferences in this format:
+
+```md
+ "theme": "light",
+ "language": "English",
+ "notifications": "enabled"
+```
+
+1. You should define a function named `add_setting` that takes a dictionary of settings and a tuple containing a key-value pair as parameters.
+
+1. `add_setting(settings, (key, value))` should:
+
+   - Convert `key` to lowercase.
+
+   - If the `key` setting exists, return `Setting 'key' already exists! Cannot add a new setting with this name.`
+   
+   - If the `key` setting doesn't exist, add the key-value pair to the `settings` dictionary and return `Setting 'key' added with value 'value' successfully!`
+
+1. You should define a function named `update_setting` that takes the a dictionary of settings and a tuple containing a key-value pair as parameters.
+
+1. `update_setting(settings, (key, value))` should:
+    
+   - Convert `key` to lowercase.
+   
+   - If the `key` setting exists, update its value in the `settings` dictionary and return: `Setting 'key' updated to 'value' successfully!`
+   
+   - If the `key` setting doesn't exist, return `Setting 'key' does not exist! Cannot update a non-existing setting.`
+
+1. You should define a function named `delete_setting` that takes the a dictionary of settings and a key as parameters.
+
+1. `delete_setting(settings, key)` should:
+    
+   - Convert `key` to lowercase.
+   
+   - If the `key` setting exists, remove the key-value pair from `settings` and return `Setting 'key' deleted successfully!`
+   
+   - If the `key` setting does not exist, return `Setting not found!`
+
+1. You should define a function named `view_settings` that takes a dictionary of settings as argument.
+
+1. `view_settings(settings)` should:
+   
+   - Return `No settings available.` if the `settings` dictionary is empty.
+   
+   - If the dictionary contains any settings, return them formatted as follows:
+
+```md
+Current User Settings:
+Theme: dark
+Notifications: enabled
+Volume: high
+
+```
+
+# --hints--
+
+You should create a dictionary named `settings` with the correct values.
+
+```js
+assert.match(code, /settings\s*=\s*{[^}]*}/)
+```
+
+You should define a function named `add_setting`.
```

Comment:
This passes by only creating the `settings` dictionary.

<img width="953" alt="Screenshot 2025-06-16 at 9 37 23 AM" src="https://github.com/user-attachments/assets/7227ca61-da04-4913-9a86-6c7438d86541" />